src/handlers.rs

```diff
@@ -909,6 +912,13 @@ pub(crate) async fn handle_command(cli_opts: CliOpts) -> Result<String, Error> {
             }
             Ok("".to_string())
         }
+        CliSubCommand::Descriptor(args) => {
+            let network = cli_opts.network;
+            let descriptor = generate_descriptor_from_args(args.clone(), network)
+                .map_err(|e| SerdeError::custom(e.to_string()))?;
```

Comment:
You don't need to pass the Generic error inside Serde json 

 Instead of `Serde json error: Generic error:`,  you can do `Generic error: ....` directly